apps/managedidentity/managedidentity.go

```diff
@@ -76,59 +79,154 @@ func (s systemAssignedValue) value() string { return string(s) }
 func (c ClientID) value() string            { return string(c) }
 func (o ObjectID) value() string            { return string(o) }
 func (r ResourceID) value() string          { return string(r) }
+func SystemAssigned() ID {
+	return systemAssignedValue("")
+}
+
+// Client is a client that provides access to Managed Identity token calls.
+type Client struct {
+	// cacheAccessorMu *sync.RWMutex
+	httpClient ops.HTTPClient
+	MiType     ID
```

Comment:
Applications can change this at any time because it's exported. This client would be more consistent with the confidential client and I expect easier to reason about if the ID were settable only at construction.